core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/V1WorkspaceMigrationMonitor.scala

```diff
@@ -0,0 +1,121 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import cats.implicits.{catsSyntaxOptionId, toTraverseOps}
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{ReadAction, WriteAction}
+import org.broadinstitute.dsde.rawls.model.Workspace
+import org.broadinstitute.dsde.rawls.monitor.MigrationOutcome.{Failure, Success}
+import org.conscrypt.ct.SerializationException
+
+import java.time.LocalDateTime
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+sealed trait MigrationOutcome
+
+object MigrationOutcome {
+  case object Success extends MigrationOutcome
+
+  final case class Failure(message: String) extends MigrationOutcome
+}
+
+final case class V1WorkspaceMigrationAttempt(id: Long,
+                                             workspaceId: UUID,
+                                             created: LocalDateTime,
+                                             started: Option[LocalDateTime],
+                                             finished: Option[LocalDateTime],
+                                             outcome: Option[MigrationOutcome])
+
+object V1WorkspaceMigrationAttempt {
+  type RecordType = (Long, UUID, LocalDateTime, Option[LocalDateTime], Option[LocalDateTime], Option[String], Option[String])
+
+  def fromRecord(record: RecordType): Either[String, V1WorkspaceMigrationAttempt] = {
+    type EitherStringT[T] = Either[String, T]
+    record match {
+      case (id, workspaceId, created, started, finished, outcome, message) =>
+        outcome
+          .traverse[EitherStringT, MigrationOutcome] {
+            case "Success" => Right(Success)
+            case "Failure" => Right(Failure(message.getOrElse("")))
+            case other => Left(s"""Failed to read V1WorkspaceMigrationAttempt from record.
+                                   | Unknown migration outcome -- "$other"""".stripMargin)
+          }
+          .map {
+            V1WorkspaceMigrationAttempt(id, workspaceId, created, started, finished, _)
+          }
+    }
+  }
+
+  def unsafeFromRecord(record: RecordType): V1WorkspaceMigrationAttempt =
+    fromRecord(record) match {
+      case Right(r) => r
+      case Left(msg) => throw new SerializationException(msg)
+    }
+
+  def toRecord(attempt: V1WorkspaceMigrationAttempt): RecordType = {
+    val (outcome: Option[String], message: Option[String]) = attempt.outcome match {
+      case Some(Success) => ("Success".some, None)
+      case Some(Failure(message)) => ("Failure".some, message.some)
+      case None => (None, None)
+    }
+    (attempt.id, attempt.workspaceId, attempt.created, attempt.started, attempt.finished, outcome, message)
+  }
+}
+
+trait V1WorkspaceMigrationComponent {
```

Comment:
I'm not sure about including the slick stuff here in the monitor, why don't we want a separate `V1WorkspaceMigrationComponent.scala` with the other tables?

Reply:
I've included it here because this is a prototype. I also don't want knowledge of the table to extend past the monitor/actor. If we need to move it out, we can later - I think it helps having it in the same file for now.

Reply:
I actually like it in the same file because it makes the ownership of the table clear, personally.